File: Project Euler/Problem 4 - Largest Palindrome Product/Problem 4 - Largest Palindrome Product.py
def palindrome(i):
    return str(i) == str(i)[::-1]


largest = 0

for a in range(1, 100):
    for b in range(1, 100):
        num = a * b
        if palindrome(num) and num > largest:
            largest = num

print(largest)

File: Project Euler/Problem 4 - Largest Palindrome Product/Problem 4 - Largest Palindrome Product.c
#include <stdio.h>
#include <stdlib.h>
#pragma warning(disable:4996)


void main() {

	int num;
	int k;
	int temp;
	int a, b;
	int largest = 1;

	for (a = 1; a < 1000; a++) {
		for (b = 1; b < 1000; b++) {
			k = 0;
			num = a * b;
			temp = num;
			while (temp > 0) {
				k = k * 10;
				k = k + temp % 10;
				temp = temp / 10;
			}

			if (k == num && k > largest) {
				largest = k;
			}
		}
		
			
	}

	printf("largest palindrome number is %d\n", largest);
	
	
	

	


	system("pause");
}
